modules/gdprEnforcement.js

```diff
@@ -31,18 +38,21 @@ function getGvlid() {
 
 /**
  * This function takes in rules and consentData as input and validates against the consentData provided. If it returns true Prebid will allow the next call else it will log a warning
- * @param {Object} rules enforcement rules set in config
- * @param {Object} consentData gdpr consent data
+ * @param {Object} rule - enforcement rules set in config
+ * @param {Object} consentData - gdpr consent data
+ * @param {number} purpose - Defines which purpose (1, 2, 4, 7) is under check
+ * @param {string=} currentModule - Bidder code of the current module
+ * @param {number=} gvlid - GVL ID for the module
  * @returns {boolean}
  */
-function validateRules(rule, consentData, currentModule, gvlid) {
+function validateRules(rule, consentData, purpose, currentModule, gvlid) {
   let isAllowed = false;
   if (!rule.vendorExceptions) rule.vendorExceptions = [];
-  if (rule.enforcePurpose && rule.enforceVendor) {
```

Comment:
I was just about to make a code change to fix this issue, where if one of these was set to false it never got in the checks!

Looks like this is the way to go instead of first checking for existence of the keys themselves.

Thanks!